sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +744,51 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, p):
```

Comment:
I would suggest using `m` instead of `p` to indicate that the modulus need not be a prime. Then it would be possible to denote the prime factors of the modulus by `p` instead of `b`. That would make the code easier to review.

Reply:
Done.